examples/flutter_gallery/lib/demo/rally/charts/line_chart.dart

```diff
@@ -0,0 +1,198 @@
+    return CustomPaint(painter: RallyLineChartPainter(context, events));
+  }
+}
+
+class RallyLineChartPainter extends CustomPainter {
+  RallyLineChartPainter(this.context, this.events);
+
+  final BuildContext context;
+
+  // Events to plot on the line as points.
+  final List<DetailedEventData> events;
+
+  // Number of days to plot.
+  // This is hardcoded to reflect the dummy data, but would be dynamic in a real
+  // app.
+  final int numDays = 52;
+
+  // Beginning of window. The end is this plus numDays.
+  // This is hardcoded to reflect the dummy data, but would be dynamic in a real
+  // app.
+  final DateTime startDate = DateTime.utc(2018, 12, 1);
+
+  // Ranges uses to lerp the pixel points.
+  // This is hardcoded to reflect the dummy data, but would be dynamic in a real
+  // app.
+  final double maxAmount = 3000.0; // minAmount is assumed to be 0.0
+
+  // The number of milliseconds in a day. This is the inherit period fot the
+  // points in this line.
+  static const int millisInDay = 24 * 60 * 60 * 1000;
+
+  // Amount to shift the tick drawing by so that the sunday ticks do not start
+  // on the edge.
+  final int tickShift = 3;
+
+  // Arbitrary unit of space for absolute positioned painting.
+  final double space = 16.0;
+
+  @override
+  void paint(Canvas canvas, Size size) {
+    final double ticksTop = size.height - space * 5;
+    final double labelsTop = size.height - space * 2;
+    _drawLine(
+      canvas,
+      Rect.fromLTWH(0.0, 0.0, size.width, ticksTop),
+    );
+    _drawXAxisTicks(
+      canvas,
+      Rect.fromLTWH(0.0, ticksTop, size.width, labelsTop - ticksTop),
+    );
+    _drawXAxisLabels(
+      canvas,
+      Rect.fromLTWH(0.0, labelsTop, size.width, size.height - labelsTop),
+    );
+  }
+
+  @override
+  bool shouldRepaint(CustomPainter oldDelegate) {
+    return false;
+  }
+
+  void _drawLine(Canvas canvas, Rect rect) {
+    final Paint linePaint = Paint()
+      ..color = RallyColors.accountColor(2)
+      ..style = PaintingStyle.stroke
+      ..strokeWidth = 2.0;
+
+    // Arbitrary value for the first point. In a real app, a wider range of
+    // points would be used that go beyond the boundaries of the screen.
+    double lastAmount = 800.0;
+
+    // Try changing this value between 1, 7, 15, etc.
+    const int smoothing = 7;
+
+    // Align the points with equal deltas (1 day) as a cumulative sum.
+    int startMillis = startDate.millisecondsSinceEpoch;
+    final List<Offset> points = <Offset>[
+      Offset(0.0, (maxAmount - lastAmount) / maxAmount * rect.height)
+    ];
+    for (int i = 0; i < numDays + smoothing; i++) {
+      final int endMillis = startMillis + millisInDay * 1;
+      final List<DetailedEventData> filteredEvents =
+          events.where((DetailedEventData e) {
+        return startMillis <= e.date.millisecondsSinceEpoch &&
+            e.date.millisecondsSinceEpoch <= endMillis;
+      }).toList();
+      lastAmount += filteredEvents.fold<num>(
```

Comment:
1 line

Reply:
Not sure here exactly what line you're referring to but I put the lastAmount expression on one line now.

Reply:
it is ok to go over the line limit for statements like this:

return startMillis <= e.date.millisecondsSinceEpoch && e.date.millisecondsSinceEpoch <= endMillis;